dallinger/custom.py

```diff
@@ -1211,6 +1212,7 @@ def worker_function(event_type, assignment_id, participant_id):
             # If it isn't, fail their nodes and recruit a replacement.
             if not worked:
                 participant.status = "bad_data"
+                session.commit()
                 exp.data_check_failed(participant=participant)
```

Comment:
@suchow why is the commit before the data_check_failed? That function should be totally safe, the risk arises when the recruiter next tries to recruit another participant. Should the commit be 1 line down?